backend/infrahub/prefect_server/app.py

```diff
@@ -1,16 +1,47 @@
 from __future__ import annotations
 
+import asyncio
+import os
+
 from fastapi import APIRouter, FastAPI
 from prefect.server.api.server import create_app
 
 from . import events
+from .bootstrap import init_prefect
+
+GLOBAL_TASKMGR_INIT_LOCK = "global.taskmgr.init"
 
 router = APIRouter(prefix="/infrahub")
 
 router.include_router(events.router)
 
 
+async def _init_prefect() -> None:
+    # Import there in case we are running Prefect within a testsuite using the original Prefect container
+    from infrahub import lock
+    from infrahub.lock import initialize_lock
+    from infrahub.services import InfrahubServices
+    from infrahub.workers.dependencies import get_cache
+
+    cache = await get_cache()
+    service = await InfrahubServices.new(cache=cache)
+    initialize_lock(service=service)
+
+    async with lock.registry.get(name=GLOBAL_TASKMGR_INIT_LOCK):
+        await init_prefect()
+
+
 def create_infrahub_prefect() -> FastAPI:
+    if (
+        os.getenv("PREFECT_API_BLOCKS_REGISTER_ON_START") == "false"
+        and os.getenv("PREFECT_API_DATABASE_MIGRATE_ON_START") == "false"
+    ):
+        # We are probably running distributed mode
+        from infrahub import config
```

Comment:
As we need to do this inline import perhaps we should move it so that all of the inline imports are done in the same location.

Alternatively we could have something like:

```python
infrahub.prefect_server.prod:create_infrahub_prod_prefect
```

Then we could just import everything at the top. But I don't know if it causes other problems.

Either or is fine with me, just adding some random thoughts.
Then this function just imports the other one.